lib/scan/shodan/engine.py

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import socket
+import socks
+import time
+import json
+import threading
+import logging
+import requests
+from core.alert import warn, info, messages
+from lib.socks_resolver.engine import getaddrinfo
+from core.targets import target_type
+from core.targets import target_to_host
+from core._time import now
+from core.log import __log_into_file
+import shodan
+
+
+def extra_requirements_dict():
+    return {
+        "shodan_scan": ["True"],
```

Comment:
why shodan_scan=True is needed here? seems unnecessary . Should be something  like "`shodan_query_override`" - if defined it will be used instead of target